### ISSUE TITLE
[Inductor][CPU] Resnet50 training performance is 0.88x of eager

### ISSUE THREAD

**Issue by blzheng** [pytorch/torchdynamo#1988]
We observed that the performance of Resnet50 training on CPU (ICX-8380) is about 0.88x of eager. It seems to be a regression introduced by enabling the decomp of batch norm in inductor. The performance will back if we comment out [aten._native_batch_norm_legit_functional ](https://github.com/pytorch/pytorch/blob/master/torch/_inductor/decomposition.py#L70) and [aten.native_batch_norm_backward](https://github.com/pytorch/pytorch/blob/master/torch/_inductor/decomposition.py#L71) in https://github.com/pytorch/pytorch/blob/master/torch/_inductor/decomposition.py.

Reproduce command: `python -m torch.backends.xeon.run_cpu --node_id 0 --enable_jemalloc benchmarks/dynamo/torchbench.py --performance --float32 -dcpu --training --inductor  --no-skip --dashboard --only resnet50 --cold_start_latency`

**Comment by blzheng** [comment]
In our analysis of generated code, we found some computation is unnecessary. Taking the following kernel as an example, the highlighted part is not necessary because we can reuse the computation for out_ptr1 to get out_ptr4. Then combined with the corresponding python code, we can conclude that the highlighted part is correspond to `var_3 = torch.ops.aten.var.correction(convolution_1, [0, 2, 3], correction = 0)`. It is redundant computation because `var_2 = torch.ops.aten.var.correction(convolution_1, [0, 2, 3], correction = 0, keepdim = True)` already finished the same computation. Though their results are in different shapes, we can replace the computation with view operations. 

kernel_cpp_1:
![image](https://user-images.githubusercontent.com/69951214/207016590-c3993906-7e87-4195-af64-f659f27f72e8.png)
corresponding python code:
![image](https://user-images.githubusercontent.com/69951214/207021064-eaad735c-cce1-4373-a77c-52a75bd9c660.png)
